source/fundamentals/crud/compound-operations.txt

```diff
@@ -0,0 +1,309 @@
+both operations. MongoDB prevents this by placing a write lock on the
+document you are modifying for the duration of your compound operation.
+
+MongoDB supports the following compound operations:
+
+- :ref:`Find and delete one document <go-find-and-delete>`
+- :ref:`Find and update one document <go-find-and-update>`
+- :ref:`Find and replace one document <go-find-and-replace>`
+
+.. tip::
+    
+   If you need to read and write to more than one document, use
+   :manual:`transactions </core/transactions/>`.
+
+Sample Data
+~~~~~~~~~~~
+
+Run the following snippet to load the documents into the ``ratings``
+collection of the ``tea`` database:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/compoundOperations.go
+   :language: go
+   :dedent:
+   :start-after: begin insertDocs
+   :end-before: end insertDocs
+
+.. include:: /includes/fundamentals/tea-sample-data-ending.rst
+
+.. _go-find-and-delete:
+
+Find and Delete
+---------------
+
+To find and delete one document, pass a ``Context`` type and a
+``non-nil`` query filter specifying which documents to match to the
+``FindOneAndDelete()`` function. 
+
+The function optionally takes a ``FineOneAndDeleteOptions`` type as a
+third parameter, which represents options you can use to configure the
+compound operation. If you don't specify a ``FineOneAndDeleteOptions``,
+the driver uses the default values for each option.
+
+The ``FineOneAndDeleteOptions`` type allows you to configure options
+with the following functions:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Function
+     - Description
+
+   * - ``SetCollation()``
+     - | The type of language collation to use when sorting results.
+       | Default: ``nil``
+
+   * - ``SetMaxTime()``
+     - | The maximum amount of time that the query can run on the server.
+       | Default: ``nil``
+
+   * - ``SetProjection()``
+     - | The fields to include in the document returned.
+       | Default: ``nil``
+
+   * - ``SetSort()``
+     - | The sort fields and directions to order the documents matched. 
+       | Default: ``nil``
+
+   * - ``SetHint()``
+     - | The index to use to scan for documents.
+       | Default: ``nil``
+
+The ``FindOneAndDelete()`` function returns a ``SingleResult``
+containing the deleted document.
+
+Example
+~~~~~~~
+
+The following example matches and deletes a document where the ``type``
+is ``"Assam"`` with the ``FindOneAndDelete()`` function:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/compoundOperations.go
+   :language: go
+   :dedent:
+   :start-after: begin FindOneAndDelete
+   :end-before: end FindOneAndDelete
+
+After running this example, the output resembles the following:
+
+.. code-block:: none
+   :copyable: false
+
+   [{_id ObjectID("...")} {type Assam} {rating 5}]
+
+.. _go-find-and-update:
+
+Find and Update
+---------------
+
+To find and update one document, pass the following parameters in order
+to the ``FindOneAndUpdate()`` function:
```

Comment:
Same as above, more description about what `FindOneAndUpdate` does would be helpful here. Specifically, how does it differ from `UpdateOne` and does it return the found document before or after the update?